### PR TITLE
[SPARK-12959] [SQL] Writing Bucketed Data with Disabled Bucketing in SQLConf

### PR DESCRIPTION
**Pull request by gatorsmile**
When users turn off bucketing in SQLConf, we should issue some messages to tell users these operations will be converted to normal way.

Also added a test case for this scenario and fixed the helper function. 

Do you think this PR is helpful when using bucket tables? @cloud-fan Thank you!

### CODE REVIEW

File: sql/hive/src/test/scala/org/apache/spark/sql/sources/BucketedWriteSuite.scala [hunk 1]
```diff
@@ -183,4 +186,25 @@ class BucketedWriteSuite extends QueryTest with SQLTestUtils with TestHiveSingle
       }
     }
   }
+
+  test("write bucketed data with bucketing disabled") {
+    withSQLConf(SQLConf.BUCKETING_ENABLED.key -> "false") {
```
Comment (cloud-fan) [inline sql/hive/src/test/scala/org/apache/spark/sql/sources/BucketedWriteSuite.scala]: hmmm, will a user manually set `BUCKETING_ENABLED` to false and then manually specify bucket information while writing data? cc @rxin @marmbrus

**Comment by rxin** [comment]
I think the bucketing config should only impact read path, not write path?

**Comment by viirya** [comment]
yeah, looks like it is.

**Comment by gatorsmile** [comment]
Thank you! @cloud-fan @rxin @viirya . 

Then, should we change the current conf name? The current name sounds like covering both writing and reading. `spark.sql.sources.bucketedTable.read` -> `spark.sql.sources.bucketedTable.read`？I am not good at naming.

**Comment by cloud-fan** [comment]
No, it will impact write path: https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala#L466 and https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/WriterContainer.scala#L314

**Comment by gatorsmile** [comment]
Yeah, I am learning your PR. @cloud-fan  I am thinking if we should have two functions for `getBucketSpec`? Something like `getBucketReadingSpec` and `getBucketWritingSpec `?

**Comment by cloud-fan** [comment]
I think this config should not impact write path logically, how about we use `maybeBucketSpec` instead of `getBucketSpec` in write path?

**Comment by gatorsmile** [comment]
Yeah, that is a good idea! Will make a try

**Comment by cloud-fan** [comment]
LGTM

**Comment by viirya** [comment]
LGTM

**Comment by rxin** [comment]
Thanks - going to merge this.

**Comment by gatorsmile** [comment]
Thank you for your reviews!

**Pull request closed without merging**